parser/func.c

```diff
@@ -53,10 +54,12 @@ apply_result *func_def_apply(parser_node *node, context *ctx)
         add_text(ctx, "mov %s, %s", sym->repl, regname);
     }
 
-    for (int i = 0; i < func->num_statements; i++)
-    {
-        parser_node *node = func->statements[i];
-        node->apply(node, ctx);
+    if (func->statements) {
```

Comment:
This is not needed. When code reaches here we already know `func->statements` is not null